cli/command/image/trust.go

```diff
@@ -37,12 +38,14 @@ func TrustedPush(ctx context.Context, cli command.Cli, repoInfo *registry.Reposi
 
 	defer responseBody.Close()
 
-	return PushTrustedReference(cli, repoInfo, ref, authConfig, responseBody)
+	return PushTrustedReference(cli, repoInfo, ref, authConfig, cli.ConfigFile().ParseTrustPinning(), responseBody)
 }
 
 // PushTrustedReference pushes a canonical reference to the trust server.
 // nolint: gocyclo
-func PushTrustedReference(streams command.Streams, repoInfo *registry.RepositoryInfo, ref reference.Named, authConfig types.AuthConfig, in io.Reader) error {
+func PushTrustedReference(streams command.Streams, repoInfo *registry.RepositoryInfo, ref reference.Named,
+	authConfig types.AuthConfig, trustPinningConfig trustpinning.TrustPinConfig, in io.Reader,
```

Comment:
The list of arguments here is becoming a bit lengthy; might be worth looking if we should refactor this (also see my other comment)